File: example/Simplace_Soil_Temperature/src/cpp/Simplace_Soil_Temperature/SoilTemperatureComponent.cpp
#include "SoilTemperatureComponent.h"
using namespace Simplace_Soil_Temperature;
SoilTemperatureComponent::SoilTemperatureComponent()
{
       
}


double SoilTemperatureComponent::getcCarbonContent(){ return this->cCarbonContent; }
double SoilTemperatureComponent::getcAlbedo(){ return this->cAlbedo; }
std::vector<double> & SoilTemperatureComponent::getcSoilLayerDepth(){ return this->cSoilLayerDepth; }
double SoilTemperatureComponent::getcFirstDayMeanTemp(){ return this->cFirstDayMeanTemp; }
double SoilTemperatureComponent::getcAverageGroundTemperature(){ return this->cAverageGroundTemperature; }
double SoilTemperatureComponent::getcAverageBulkDensity(){ return this->cAverageBulkDensity; }
double SoilTemperatureComponent::getcDampingDepth(){ return this->cDampingDepth; }

void SoilTemperatureComponent::setcCarbonContent(double _cCarbonContent)
{
    _SnowCoverCalculator.setcCarbonContent(_cCarbonContent);
}
void SoilTemperatureComponent::setcAlbedo(double _cAlbedo)
{
    _SnowCoverCalculator.setAlbedo(_cAlbedo);
}
void SoilTemperatureComponent::setcSoilLayerDepth(const std::vector<double> & _cSoilLayerDepth)
{
    _STMPsimCalculator.setcSoilLayerDepth(_cSoilLayerDepth);
}
void SoilTemperatureComponent::setcFirstDayMeanTemp(double _cFirstDayMeanTemp)
{
    _STMPsimCalculator.setcFirstDayMeanTemp(_cFirstDayMeanTemp);
}
void SoilTemperatureComponent::setcAverageGroundTemperature(double _cAverageGroundTemperature)
{
    _STMPsimCalculator.setcAVT(_cAverageGroundTemperature);
}
void SoilTemperatureComponent::setcAverageBulkDensity(double _cAverageBulkDensity)
{
    _STMPsimCalculator.setcABD(_cAverageBulkDensity);
}
void SoilTemperatureComponent::setcDampingDepth(double _cDampingDepth)
{
    _STMPsimCalculator.setcDampingDepth(_cDampingDepth);
}
void SoilTemperatureComponent::Calculate_Model(SoilTemperatureState &s, SoilTemperatureState &s1, SoilTemperatureRate &r, SoilTemperatureAuxiliary &a, SoilTemperatureExogenous &ex)
{
    ex.setiTempMax(ex.getiAirTemperatureMax());
    ex.setiTempMin(ex.getiAirTemperatureMin());
    ex.setiRadiation(ex.getiGlobalSolarRadiation());
    ex.setiSoilTempArray(s.getSoilTempArray());
    _SnowCoverCalculator.Calculate_Model(s, s1, r, a, ex);
    ex.setiSoilSurfaceTemperature(s.getSoilSurfaceTemperature());
    _STMPsimCalculator.Calculate_Model(s, s1, r, a, ex);
}
SoilTemperatureComponent::SoilTemperatureComponent(SoilTemperatureComponent& toCopy)
{
    cCarbonContent = toCopy.getcCarbonContent();
    cAlbedo = toCopy.getcAlbedo();
    for (int i = 0; i < toCopy.getcSoilLayerDepth().size(); i++)
{
    cSoilLayerDepth[i] = toCopy.getcSoilLayerDepth()[i];
}

    cFirstDayMeanTemp = toCopy.getcFirstDayMeanTemp();
    cAverageGroundTemperature = toCopy.getcAverageGroundTemperature();
    cAverageBulkDensity = toCopy.getcAverageBulkDensity();
    cDampingDepth = toCopy.getcDampingDepth();
}

File: example/Simplace_Soil_Temperature/src/java/Simplace_Soil_Temperature/SoilTemperatureComponent.java
public class SoilTemperatureComponent
{
    
    public SoilTemperatureComponent() { }

    SnowCoverCalculator _SnowCoverCalculator = new SnowCoverCalculator();
    STMPsimCalculator _STMPsimCalculator = new STMPsimCalculator();

    public double getcCarbonContent()
    { return _SnowCoverCalculator.getcCarbonContent(); }
    public void setcCarbonContent(double _cCarbonContent){
    _SnowCoverCalculator.setcCarbonContent(_cCarbonContent);
    }

    public double getcAlbedo()
    { return _SnowCoverCalculator.getAlbedo(); }
    public void setcAlbedo(double _cAlbedo){
    _SnowCoverCalculator.setAlbedo(_cAlbedo);
    }

    public Double [] getcSoilLayerDepth()
    { return _STMPsimCalculator.getcSoilLayerDepth(); }
    public void setcSoilLayerDepth(Double [] _cSoilLayerDepth){
    _STMPsimCalculator.setcSoilLayerDepth(_cSoilLayerDepth);
    }

    public double getcFirstDayMeanTemp()
    { return _STMPsimCalculator.getcFirstDayMeanTemp(); }
    public void setcFirstDayMeanTemp(double _cFirstDayMeanTemp){
    _STMPsimCalculator.setcFirstDayMeanTemp(_cFirstDayMeanTemp);
    }

    public double getcAverageGroundTemperature()
    { return _STMPsimCalculator.getcAVT(); }
    public void setcAverageGroundTemperature(double _cAverageGroundTemperature){
    _STMPsimCalculator.setcAVT(_cAverageGroundTemperature);
    }

    public double getcAverageBulkDensity()
    { return _STMPsimCalculator.getcABD(); }
    public void setcAverageBulkDensity(double _cAverageBulkDensity){
    _STMPsimCalculator.setcABD(_cAverageBulkDensity);
    }

    public double getcDampingDepth()
    { return _STMPsimCalculator.getcDampingDepth(); }
    public void setcDampingDepth(double _cDampingDepth){
    _STMPsimCalculator.setcDampingDepth(_cDampingDepth);
    }
    public void  Calculate_Model(SoilTemperatureState s, SoilTemperatureState s1, SoilTemperatureRate r, SoilTemperatureAuxiliary a, SoilTemperatureExogenous ex)
    {
        ex.setiTempMax(ex.getiAirTemperatureMax());
        ex.setiTempMin(ex.getiAirTemperatureMin());
        ex.setiRadiation(ex.getiGlobalSolarRadiation());
        ex.setiSoilTempArray(s.getSoilTempArray());
        _SnowCoverCalculator.Calculate_Model(s, s1, r, a, ex);
        ex.setiSoilSurfaceTemperature(s.getSoilSurfaceTemperature());
        _STMPsimCalculator.Calculate_Model(s, s1, r, a, ex);
    }
    private double cCarbonContent;
    private double cAlbedo;
    private double Albedo;
    private Double [] cSoilLayerDepth;
    private double cFirstDayMeanTemp;
    private double cAverageGroundTemperature;
    private double cAVT;
    private double cAverageBulkDensity;
    private double cABD;
    private double cDampingDepth;
    public SoilTemperatureComponent(SoilTemperatureComponent toCopy) // copy constructor 
    {
        this.cCarbonContent = toCopy.getcCarbonContent();
        this.cAlbedo = toCopy.getcAlbedo();
        
        for (int i = 0; i < toCopy.getcSoilLayerDepth().length; i++)
        {
            cSoilLayerDepth[i] = toCopy.getcSoilLayerDepth()[i];
        }
        this.cFirstDayMeanTemp = toCopy.getcFirstDayMeanTemp();
        this.cAverageGroundTemperature = toCopy.getcAverageGroundTemperature();
        this.cAverageBulkDensity = toCopy.getcAverageBulkDensity();
        this.cDampingDepth = toCopy.getcDampingDepth();

    }
}